posts/python-in-production-part-3-of-5.md

```diff
@@ -0,0 +1,107 @@
+
+### Keep your Virtual Environments Separate
+
+Having multiple virtual environments helps to keep things clean.  Notice how in the build script we activate the virtual environment that lives in the dist folder, but we get the site packages for our binary from our development virtual environment.  That is because the dependencies required to run shiv are different then the ones required to build our program.
+
+### A Note About Building Remotely
+
+In a subsequent part of this series, we will put this process into a Jenkins build server.  When that happens, it will be necessary to create a separate bash script for setting up the initial virtual environment (refered to as the development virtual environment above) and installing our programs dependencies, because it will not exist on a newly provisioned machine.
+
+# Running the Executable Binary
+
+Congratulations! You have just created a single file, binary executable.  In order to run the binary, we must go back to the terminal.
+```bash
+./simple_api
+```
+I encourage you to put this executable binary onto a different machine and run it!  As long as that machine has the proper version of Python installed, the program will run.  
+
+### CPU Architecture Matters
+
+There are a few things worth mentioning that could cause you some errors along the way, especially if your program is not *pure python*.  If you build this binary on a machine running an ARM processor, and then copy it to a machine running an AMD processor, you may run into some issues.  In a later part of the series, we will discuss building for different architectures in more depth.  Remember from part 1, I mentioned that it is good to know which version of python you are building for?  Well, it is also good to know which architecture you are building for too!  
+
+Some packages that are architecture specific may include things like Audio Controllers, Systemd bindings, or SPI dev kits.  Mainly because these packages interact with the hardware of the computer and therefore are more architecture specific.
+
+# Update your .gitignore and  Push Your Changes
+
+Now that we are building an executable, we have lots of *artifacts* in our repo.  For instance `simple_api.egg-info` is not something we want in our source control repository.  There is a really awesome template `.gitignore` file [here](https://github.com/github/gitignore/blob/main/Python.gitignore).  This is what I will be using moving forward.
+
+In [part 4](https://temporarylink) of this series, we will load our binary onto a linux system and run it using systemd.  We will learn how to create a systemd unit file, pipe logging to journald, and how to use an Environment File to configure our program.
```

Comment:
At the end of your blog, include a call to action. Something along the lines of 
`For more information on how Ippon Technologies can help your organization utilize Python, contact sales@ipponusa.com.`